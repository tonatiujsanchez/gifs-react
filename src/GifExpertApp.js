import { useState } from "react"
import AddCategory from "./components/AddCategory"
import GifGrid from "./components/GifGrid"



const GifExpertApp = ({ defaultCategories = [] }) => {

    const [ categories, setCategories ] = useState(defaultCategories)

    const addNewCategory = ( newCategory ) =>{
        setCategories( [ newCategory, ...categories ] )
    }


    return (
      <>
        <h1 className="title">Gifs App</h1>
        <AddCategory addNewCategory={ addNewCategory } />
        <ol>
            {
                categories.map( category => (
                    <GifGrid key={ category } category={ category } />
                    ))
            }
        </ol>
      </>
    )
  }
  
  export default GifExpertApp