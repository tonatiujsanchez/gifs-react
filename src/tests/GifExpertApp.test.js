import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"



describe('Pruebas al compoenente principal GifExpertApp', () => { 


    const wrapper = shallow( <GifExpertApp /> )

    test('El componente GifExpertApp debe de mostrarse correctamente', () => { 
        expect( wrapper ).toMatchSnapshot()
     })



     test('Debe mostrar una lista de categorias', () => { 

        const defaultCategories = ['Dragon Ball', 'One Punch']

        
        const wrapper = shallow( <GifExpertApp defaultCategories={ defaultCategories } /> )
        expect( wrapper ).toMatchSnapshot()


        const gifGrid = wrapper.find('GifGrid')
        expect( gifGrid.length ).toBe( defaultCategories.length )
        
      })


 })