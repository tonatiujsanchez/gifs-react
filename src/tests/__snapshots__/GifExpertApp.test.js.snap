// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al compoenente principal GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1
    className="title"
  >
    Gifs App
  </h1>
  <AddCategory
    addNewCategory={[Function]}
  />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al compoenente principal GifExpertApp El componente GifExpertApp debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1
    className="title"
  >
    Gifs App
  </h1>
  <AddCategory
    addNewCategory={[Function]}
  />
  <ol />
</Fragment>
`;
